ports/stm32f4/common-hal/microcontroller/Pin.c

```diff
@@ -54,33 +54,39 @@ void reset_all_pins(void) {
 
 // Mark pin as free and return it to a quiescent state.
 void reset_pin_number(uint8_t pin_port, uint8_t pin_number) {
-    if (pin_number == NO_PIN) {
+    if (pin_port == 0x00) {
```

Comment:
Isn't pin_port = 0 valid? Typically I use a high value to indicate it is invalid.

Reply:
Whoops I got that mixed up with the ADC. It'd be 0x0F with the bit packing, right?

Reply:
How many ports can an STM have? If it's 16 we'll need a separate bit for validity.